doc/sphinx/addendum/extraction.rst

```diff
@@ -399,6 +399,109 @@ As an example of translation to a non-inductive datatype, let's turn
 
    Extract Inductive nat => int [ "0" "succ" ] "(fun fO fS n -> if n=0 then fO () else fS (n-1))".
 
+Generating FFI Code
+~~~~~~~~~~~~~~~~~~~
+
+The plugin provides mechanisms to generate (currently only) OCaml code to
+interface the generated OCaml code with C programs. In order to link compiled
+OCaml code with C code, the linker needs to know
+
+ * which C functions will be called by the ML code (externals)
+ * which ML functions shall be accessible by the C code (callbacks)
+
+.. cmd:: Extract External Constant @qualid => @string
+
+   Same as the :cmd:`Extract Constant`, except that the given ML terms
+   will not be declared via ``let`` but in the form
+   ``external @qualid : ML type = "@string"``.
+   Also, no ``val`` declaration will be generated in the respective
+   mli file.
+
+.. caution::
+
+   * The external function name provided by the user as :n:`@string`
+     is currently **not** checked at all by extraction, even for syntax errors.
+
+   * It is the responsibility of the user to ensure that the C
+     functions given to realize the axioms do have the expected or compatible
+     types. In fact, the strings containing realizing code are just copied to the
+     extracted files.
+
+   * A previously used :cmd:`Extract Constant` for :n:`@qualid` will be overridden
+     by the command :cmd:`Extract External Constant`, although internally, the
+     specified :n:`@qualid` will still be part of the (inline) constant extraction
+     table. This is by design, since, when defining a function to be external, a
+     ``val`` declaration and ``let`` definition would collide with the external declaration
+     during compilation. Also, an external function cannot be inlined. Thus, the
+     command :cmd:`Extract Inlined Constant` has to be overridden, too.
```

Comment:
Does the user actually care?

In fact, I am not even sure what the point of this silent behavior is. Would it not be better to emit an error if the extraction scheme for a given constant suddenly changes?

Reply:
Here, I did not want to be too rude to the user and rather warn in the documentation. But if it is okay to be hard and handle that as error, it is okay for me. I adopted the code accordingly.